### DISCUSSION TITLE
🎉 S2 2.0 版本开始内测啦 !!!

### DISCUSSION THREAD

**Discussion by lijinke666** [antvis/S2#1933] [Q&A]
了解 2.0 版本新特性 - [S2 2.0 表格看数新纪元](https://www.yuque.com/antv/blog/1122_7_s2) 和 [roadmap](https://github.com/antvis/S2/issues/2153)

## 🏠 官网地址变化

原官网 <https://s2.antv.vision> 和 <https://antv-s2.gitee.io> 不再维护和使用，请使用最新的文档，确保您看到的不是过时的文档。

- 原 `v1` 官网迁移至 <https://s2-v1.antv.antgroup.com>.
- 原 <https://s2.antv.antgroup.com> 将作为 `v2` 默认官网。

## 🏷️ npm dist-tag 变化

> 什么是 [dist-tag](https://docs.npmjs.com/adding-dist-tags-to-packages/) ?

`S2 2.0` 版本目前处于**内测阶段**, 部分 API 可能会根据合理性随时改动。

`npm` 的 [`dist-tag`](https://docs.npmjs.com/cli/v10/commands/npm-dist-tag) 对应关系如下：

- `@antv/s2@next` 对应 `2.x` 版本
- `@antv/s2@latest` 对应 `1.x` 版本

## 📅 正式版本发布时间

目前 `next` 会持续内测一段时间，https://s2.antv.antgroup.com 会作为相应的文档网站。

在此期间，会持续根据用户的反馈进行 Bug fix 和代码调整，在 `@antv/s2@next` 版本稳定后，会发布正式版本, `latest` 将默认指向 `2.x` 版本，去除 `next` 标识。

## 🗓️ v1 版本维护期

目前 `v1` 版本会继续维护，针对 `BUG` 发布 `Patch` 版本修复，但不再接收新的 `Feature Request` 和 `Feature Pull Request`，欢迎 `Bug Fix Pull Request`, 截止日期为 `2024` 年年底。

## 📦 安装

| Package  | Version | Size   | Download     |
| - | - | - | - |
| [@antv/s2](https://github.com/antvis/S2/tree/next/packages/s2-core) |  ![next](https://img.shields.io/npm/v/@antv/s2/next.svg)  | ![size](https://img.badgesize.io/https:/unpkg.com/@antv/s2@next/dist/index.min.js?label=gzip%20size&compression=gzip)       | ![download](https://img.shields.io/npm/dm/@antv/s2.svg)       |
| [@antv/s2-react](https://github.com/antvis/S2/tree/next/packages/s2-react) | ![next](https://img.shields.io/npm/v/@antv/s2-react/next.svg)| ![size](https://img.badgesize.io/https:/unpkg.com/@antv/s2-react@next/dist/index.min.js?label=gzip%20size&compression=gzip) | ![download](https://img.shields.io/npm/dm/@antv/s2-react.svg) |
| [@antv/s2-vue](https://github.com/antvis/S2/tree/next/packages/s2-vue) | ![next](https://img.shields.io/npm/v/@antv/s2-vue/next.svg) | ![size](https://img.badgesize.io/https:/unpkg.com/@antv/s2-vue@next/dist/index.min.js?label=gzip%20size&compression=gzip)   | ![download](https://img.shields.io/npm/dm/@antv/s2-vue.svg)   |

```bash
# 基础版本
yarn add @antv/s2@next --save

# React 版本
yarn add @antv/s2-react@next antd @ant-design/icons --save

# Vue3 版本
yarn add @antv/s2-vue@next ant-design-vue@3.x --save
```

## `@antv/s2-react` 支持 React 18 和 Ant Design 5.0

`@antv/s2-react` 的 `2.x` 版本适配了 `React 18`, 并兼容 `React 16 和 17`, 分析组件升级到了 `antd@5.x`.

### Ant Design 多版本共存

对于项目使用的是 `antd@4.x`, 或者所依赖的其他库依赖 `antd@4.x`, 由于种种历史原因无法升级到 `antd@5.x` 的情况，可以通过 [多版本共存](https://ant-design.antgroup.com/docs/react/migration-v5-cn#%E5%A4%9A%E7%89%88%E6%9C%AC%E5%85%B1%E5%AD%98) 的方式来临时过渡。

```json
// $ npm install --save antd-v5@npm:antd@5
{
  "antd": "4.x",
  "antd-v5": "npm:antd@5"
}
```

通过 Webpack 内置插件 [`NormalModuleReplacementPlugin`](https://webpack.js.org/plugins/normal-module-replacement-plugin/) 或者 `自定义 Webpack 插件` 的方式指定 `@antv/s2-react` 使用 `antd-v5`, 无需做任何修改，项目中其他依赖将继续使用 `antd@4.x`.

:::warning{title="注意"}
其他打包工具 （如 `Vite`) 或者基于 `Webpack` 封装的库或框架（如 `father`, `umi`) 同理，请自行搜索，这里不再赘述。
需要注意的是：这种方式为临时过渡解决方案，从长远来看，**[Ant Design v4 版本已于 2023 年年底停止维护](https://ant-design.antgroup.com/docs/blog/v4-ood-cn)，建议尽快升级至 `antd@5.x`.**
:::

自定义 Webpack 插件参考：

```ts
class AntdV5AliasPlugin {
  apply(compiler) {
    compiler.hooks.normalModuleFactory.tap("AntdV5AliasPlugin", (nmf) => {
      nmf.hooks.beforeResolve.tapAsync("AntdV5AliasPlugin", (resolveData, callback) => {
        if (resolveData.contextInfo?.issuer?.includes('node_modules/@antv/s2-react')) {
          // 匹配："antd" 和 "antd/es/locale/xxx"
          if (/antd(\/*)?/.test(resolveData.request)) {
            // 替换为："antd-v5" 和 "antd-v5/es/locale/xxx"
            resolveData.request = resolveData.request.replace(/antd(\/*)?/,'antd-v5$1')
          }
        }

        callback();
      });
    });
  }
}
```

欢迎尝鲜, 有任何问题欢迎在讨论区或 issue 区留言评论.

**Comment by jeefeng** [comment]
希望能支持表头和表格内容自动换行的功能, 跪求~~~

**Reply by wozien** [reply]
+1

**Reply by lijinke666** [reply]
已支持 https://github.com/antvis/S2/pull/2383

**Comment by oliver34** [comment]
啥时候发布啊，现在用next版本没有文档不知道咋搞啊

**Reply by LaraineR** [reply]
+1

**Comment by G0wtherZ** [comment]
是推迟发布了吗，有没有进一步消息？

**Comment by iLoongDD** [comment]
最近正好项目中需要，也遇到一些需求，发现2.0的版本更满足功能。想知道2.0版本什么时候发布，如果两个月内的话，我正好先试用next版本。

**Comment by lijinke666** [comment]
通告更新了, 相关疑惑请查看置顶消息.

**Comment by QJerryxjh** [comment]
想要使用s2-react2.x还要把antd到5.x啊 :anguished:

**Reply by lijinke666** [reply]
是的

**Reply by zhangming9502** [reply]
+1

**Comment by RJM1996** [comment]
升级2.0必须还要升级 antd 到 5.0 吗 
<img width="634" alt="Snipaste_2024-03-19_14-47-46" src="https://github.com/antvis/S2/assets/33800707/38225946-7c8e-43e4-a608-9d82e6637da0">

**Reply by lijinke666** [reply]
是的

**Reply by zhangming9502** [reply]
+1

**Comment by liqiuqiui** [comment]
正式版本什么时候发布o

**Comment by lijinke666** [comment]
https://github.com/antvis/S2/issues/2986
